common/src/main/java/com/google/auto/common/SuperficialValidation.java

```diff
@@ -63,10 +64,13 @@ public static boolean validateElements(Iterable<? extends Element> elements) {
         }
 
         @Override public Boolean visitType(TypeElement e, Void p) {
+          TypeMirror superclass = e.getSuperclass();
           return isValidBaseElement(e)
               && validateElements(e.getTypeParameters())
               && validateTypes(e.getInterfaces())
-              && validateType(e.getSuperclass());
+              && validateType(superclass)
+              && validateElements(e.getInterfaces().stream().map(MoreTypes::asElement).collect(Collectors.toList()))
```

Comment:
Does `e.getInterfaces().stream(MoreTypes::asElement).allMatch(this::validateElement)`
 work?

Reply:
I'll try that.